implementations/micrometer-registry-influx/src/test/java/io/micrometer/influx/InfluxMeterRegistryVersionsTest.java

```diff
@@ -0,0 +1,395 @@
+import ru.lanwen.wiremock.ext.WiremockResolver;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Unit tests for check compatibility of {@link InfluxMeterRegistry} with InfluxDB v1 and InfluxDB v2.
+ *
+ * @author Jakub Bednar (19/05/2020 09:00)
+ */
+@ExtendWith(WiremockResolver.class)
+public class InfluxMeterRegistryVersionsTest {
+
+    @Test
+    void writeToV1Token(@WiremockResolver.Wiremock WireMockServer server) {
+        server.stubFor(any(urlPathEqualTo("/write"))
+                .willReturn(aResponse().withStatus(204)));
+
+        publishSimpleStat(InfluxApiVersion.V1, server);
+
+        server.verify(postRequestedFor(urlEqualTo("/write?consistency=one&precision=ms&db=my-db"))
+                .withRequestBody(equalTo("my_counter,metric_type=counter value=0 1"))
+                .withHeader("Authorization", equalTo("Bearer my-token")));
+    }
+
+    @Test
+    void writeToV2Token(@WiremockResolver.Wiremock WireMockServer server) {
+        server.stubFor(any(urlPathEqualTo("/api/v2/write"))
+                .willReturn(aResponse().withStatus(204)));
+
+        publishSimpleStat(InfluxApiVersion.V2, server);
+
+        server.verify(postRequestedFor(urlEqualTo("/api/v2/write?&precision=ms&bucket=my-bucket&org=my-org"))
+                .withRequestBody(equalTo("my_counter,metric_type=counter value=0 1"))
+                .withHeader("Authorization", equalTo("Token my-token")));
+    }
+
+    @Test
+    void writeToV1WithRP(@WiremockResolver.Wiremock WireMockServer server) {
+        Map<String, String> props = new HashMap<>();
+        InfluxConfig config = props::get;
+        props.put("influx.uri", server.baseUrl());
+        props.put("influx.db", "my-db");
+        props.put("influx.token", "my-token");
+        props.put("influx.retentionPolicy", "one_day_only");
+
+        publishSimpleStat(config);
+
+        server.verify(postRequestedFor(urlEqualTo("/write?consistency=one&precision=ms&db=my-db&rp=one_day_only"))
+                .withRequestBody(equalTo("my_counter,metric_type=counter value=0 1"))
+                .withHeader("Authorization", equalTo("Bearer my-token")));
+    }
+
+    @Test
+    void writeToV2TokenRequired(@WiremockResolver.Wiremock WireMockServer server) {
+
+        server.stubFor(any(urlPathEqualTo("/api/v2/write"))
+                .willReturn(aResponse().withStatus(204)));
+
+        InfluxConfig config = new InfluxConfig() {
```

Comment:
I would use a mock instead of implementing it.

Reply:
I don't think we mock config interfaces in other tests, so this is more in line with the code base.